pilot/cmd/pilot-agent/main.go

```diff
@@ -307,8 +308,10 @@ var (
 			}
 
 			opts := make(map[string]interface{})
-			if sdsEnabled {
-				opts["sds_uds_path"] = sdsUdsPathVar.Get()
+			// If control plane auth is mTLS and global SDS flag is turned on, set UDS path and token path
+			// for control plane SDS.
+			if controlPlaneAuthEnabled && sdsEnabled {
```

Comment:
Why ? Even if control plane auth is disabled, we still send this.

Reply:
If the control plane auth is disabled, the bootstrapping config shouldn't have tls_context set, since it only serves the traffic to the control plane, right?